Adapters/Vungle/ISVungleAdapter/ISVungleBannerAdapterRouter.m

```diff
@@ -0,0 +1,212 @@
+//
+//  ISVungleBannerAdapterRouter.m
+//  ISVungleAdapter
+//
+//  Copyright © 2020 ironSource. All rights reserved.
+//
+
+#import "ISVungleBannerAdapterRouter.h"
+#import "ISVungleAdapter.h"
+
+@interface ISVungleBannerAdapterRouter ()
+
+@property (nonatomic, strong, nullable) NSString *bidPayload;
+
+@end
+
+@implementation ISVungleBannerAdapterRouter
+
+- (instancetype)initWithPlacementID:(NSString *)placementID
+                      parentAdapter:(ISVungleAdapter *)parentAdapter
+                           delegate:(id<ISBannerAdapterDelegate>)delegate {
+    if (self = [super init]) {
+        _placementID = placementID;
+        _parentAdapter = parentAdapter;
+        _delegate = delegate;
+        _bannerState = UNKNOWN;
+        _bidPayload = nil;
+        self.bannerSize = BannerSizeRegular;
+    }
+
+    return self;
+}
+
+- (void)loadBannerAd {
+    self.bannerAd = [[VungleBanner alloc] initWithPlacementId:self.placementID size:self.bannerSize];
+    self.bannerAd.delegate = self;
+    self.bannerAd.enableRefresh = NO;
+    self.adView = [[UIView alloc] initWithFrame:[self getAdViewRect:self.bannerSize]];
+
+    if ([self.bannerAd canPlayAd]) {
+        LogInternal_Internal(@"Banner ad: %@ is loaded", self.placementID);
+        [self showBannerAd];
+        [self.delegate adapterBannerDidLoad:self.adView];
+        return;
+    }
+
+    [self.bannerAd load:self.bidPayload];
+}
+
+- (void)showBannerAd {
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [self.bannerAd presentOn:self.adView];
+    });
+    self.bannerState = SHOWING;
+}
+
+- (void)setBidPayload:(NSString * _Nullable)bidPayload {
+    self.bidPayload = bidPayload;
+}
+
+- (void)setSize:(ISBannerSize *)size {
+    self.bannerSize = [self getBannerSize:size];
+}
+
+- (void)destroy {
+    [self.adView removeFromSuperview];
+    self.adView = nil;
+    self.bannerAd.delegate = nil;
+    self.bannerAd = nil;
+}
+
+- (void)bannerAdInitSuccess {
```

Comment:
When is this called?

Reply:
@stanley-vungle This method is called when Vungle SDK is initialized successfully and all the ISVungleBannerAdapterRouters which are requesting Banner before Vungle SDK is initialized are waiting this method. Thanks a lot. :)

Reply:
Ok. Thanks